colorscad.sh

```diff
@@ -143,23 +143,23 @@ if [ "$FORMAT" = 3mf ]; then
 fi
 
 # Convert OUTPUT to a full path, because we're going to change current directory (see below)
-OUTPUT=$(cd "${OUTPUT%${OUTPUT##*/}}."; pwd)/${OUTPUT##*/}
+OUTPUT="$(cd "${OUTPUT%"${OUTPUT##*/}"}." || exit 1 ; pwd)/${OUTPUT##*/}"
 
 # Change the current dir to the input's dir, for consistent behavior.
 # That is because not all OpenSCAD versions behave the same when the input is not in the current dir;
 # in some versions 'import()' is relative to the current dir instead of the input's dir, and in
 # other versions .csg output is written relative to the input's dir, instead of the current dir.
 ORIGINAL_PWD=$(pwd)
-cd "${INPUT%${INPUT##*/}}."
+cd "${INPUT%"${INPUT##*/}"}." || exit 1
 INPUT=${INPUT##*/}
 
 # Create a temporary, unique .csg file in the input's directory.
 # It needs to be in the input's directory, because it might contain relative "import" statements.
 # On macOS, 'mktemp' cannot create a file with a given extension, so use a workaround.
 INPUT_CSG=$(
 	set -o noclobber
-	NAME=
-	until > "$NAME"; do
+	NAME=tmp.$$_${RANDOM}.csg
+	while [ -f "$NAME" ]; do
```

Comment:
While this might make shellcheck happy, it does introduce a race condition (and also makes the `-o noclobber` unnecessary). Let's see if we can come up with an alternative...

Reply:
This seems to keep shellcheck happy:
`until : > "$NAME"`
The `set -o noclobber` could be moved to just above the `until` to reduce the magic of it all a bit...

Reply:
More clear though, and appears to work on both mac and linux:
```
INPUT_CSG=$(
	until mktemp "tmp.$$_${RANDOM}_XXXXXX.csg"; do sleep 1; done
)
```

Reply:
this looks way better, i came up with a working variant of `until > "$NAME"` that makes shellcheck happy, but wanted to remove the side effect with already existing files.